### PR TITLE
Remove Parallel (Processes) in Marqo

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
remove Parallel processes in marqo

* **What is the current behavior?** (You can also link to an open issue here)
we have a `processes` parameter in marqo `add_or_replace_documents` API that uses multiprocess to index documents.
* **What is the new behavior (if this is a feature change)?**
We decide to remove the support of the multiprocess feature and remove the support of this parameter in the API, for the following reasons:

1. It blows out memory (it seems that loaded models are replicated for each process)

2. The (time) overhead from starting multiple processes makes it unviable for ~30 second HTTP requests. It only makes sense for much longer running requests (which is an unusual pattern for a REST API)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
yes, this is a breaking change

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
passes

* **Related Python client changes** (link commit/PR here)
on going

* **Related documentation changes** (link commit/PR here)
parameter processes has already been removed from the doc

* **Other information**:
not

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/tensor_search/api.py [hunk 1]
```diff
@@ -185,6 +185,10 @@ def add_or_replace_documents(
         ),
         mappings: typing.Optional[dict] = Depends(api_utils.decode_mappings)):
     """add_documents endpoint (replace existing docs with the same id)"""
+    if processes is not None:
+        raise InvalidArgError("The parameter `process` is no longer supported anymore. Please check the documentation for more information. \n"
```
Comment (farshidz) [inline src/marqo/tensor_search/api.py]: Parameter is `processes` (plural). Also I don't think backticks will be picked up in error message i.e. won't affect formatting. 

Could probably do without the line break. Error messages are generally one paragraph

File: src/marqo/tensor_search/tensor_search.py [hunk 2]
```diff
@@ -251,40 +250,12 @@ def add_documents_orchestrator(
...
-        if add_docs_params.index_name in get_cache():
-            logger.info(f'deleting cache entry for {add_docs_params.index_name} after parallel add documents')
-            del get_cache()[add_docs_params.index_name]
-
-        return results
     else:
         if batch_size < 0:
             raise errors.InvalidArgError("Batch size can't be less than 1!")
-        logger.debug(f"batch_size={batch_size} and processes={processes} - batching using a single process")
+        logger.debug(f"batch_size={batch_size} - batching using a single process")
```
Comment (farshidz) [inline src/marqo/tensor_search/tensor_search.py]: I think `batching using a single process` is superfluous now that there's no other option

File: src/marqo/tensor_search/api.py [hunk 3]
```diff
@@ -185,6 +185,10 @@ def add_or_replace_documents(
         ),
         mappings: typing.Optional[dict] = Depends(api_utils.decode_mappings)):
     """add_documents endpoint (replace existing docs with the same id)"""
+    if processes is not None:
```
Comment (farshidz) [inline src/marqo/tensor_search/api.py]: This needs a test

File: tests/tensor_search/test_parallel.py [hunk 4]
```diff
@@ -14,57 +14,5 @@ class TestAddDocumentsPara(MarqoTestCase):
...
-        
-        data = [{'text':f'something {str(i)}', '_id': str(i)} for i in range(100)]
-        try:
-            res = parallel.add_documents_mp(config=self.config, add_docs_params=AddDocsParams(
-                    index_name=self.index_name_1, docs=data, auto_refresh=True),
-                batch_size=10, processes=2)
-            raise AssertionError
-        except InternalError:
-            pass
+    pass
```
Comment (farshidz) [inline tests/tensor_search/test_parallel.py]: Why keep the empty file?

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/tensor_search/api.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/api.py]: updated

Comment (pandu-k) [reply src/marqo/tensor_search/api.py]: @farshidz We use backticks as quotation marks in our error messages, following the style found in other projects, like Meilisearch: https://www.meilisearch.com/docs/reference/errors/overview#error-format

File: tests/tensor_search/test_parallel.py (hunk 4 above)
Comment (wanliAlex) [reply tests/tensor_search/test_parallel.py]: deleted now

File: src/marqo/tensor_search/tensor_search.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search.py]: updated

File: src/marqo/tensor_search/api.py (hunk 3 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/api.py]: We just delete this error and make it easier.

**Review by farshidz — APPROVED** [review]

**Pull request merged**